presto-parquet/src/main/java/io/prestosql/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -249,6 +252,25 @@ public static Domain getDomain(Type type, long rowCount, Statistics<?> statistic
             return createDomain(type, hasNullValue, parquetTimestampStatistics);
         }
 
+        if (type instanceof TimestampType && statistics instanceof LongStatistics) {
+            TimestampType timestampType = (TimestampType) type;
+            LongStatistics longStatistics = (LongStatistics) statistics;
+
+            long min = longStatistics.getMin();
+            long max = longStatistics.getMax();
+            if (min > max) {
+                failWithCorruptionException(failOnCorruptedParquetStatistics, column, id, longStatistics);
+                return Domain.create(ValueSet.all(type), hasNullValue);
+            }
+
+            TimeUnit prestoTimestampRepresentationUnit = prestoTimestampRepresentationUnit(timestampType);
+            TimeUnit parquetTimestampRepresentationUnit = parquetTimestampRepresentationUnit(statistics.type().getOriginalType());
+            ParquetTimestampStatistics parquetTimestampStatistics = new ParquetTimestampStatistics(
+                    prestoTimestampRepresentationUnit.convert(longStatistics.getMin(), parquetTimestampRepresentationUnit),
+                    prestoTimestampRepresentationUnit.convert(longStatistics.getMax(), parquetTimestampRepresentationUnit));
```

Comment:
This will truncate when going from higher precision to lower precision. That's inconsistent with how Presto treats conversions between timestamp types. It does rounding, instead.

Reply:
Thanks for catching that. I pushed a fix if you have a change to take another look @martint